plan/logical_plan_builder.go

```diff
@@ -192,20 +197,38 @@ func extractOnCondition(conditions []expression.Expression, left LogicalPlan, ri
 	return
 }
 
+func extractTableAlias(p LogicalPlan) *model.CIStr {
```

Comment:
table alias can also be a subquery. leftPlan.Schema().Column[0].TableName can stand for the table alias.